src/meta/ActiveHostsMan.cpp

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/ActiveHostsMan.h"
+#include "meta/MetaServiceUtils.h"
+#include "meta/processors/BaseProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+ActiveHostsMan::ActiveHostsMan(int32_t intervalSeconds, int32_t expiredSeconds,
+                               kvstore::KVStore* kv)
+    : intervalSeconds_(intervalSeconds)
+    , expirationInSeconds_(expiredSeconds) {
+    if (kv != nullptr) {
+        kvstore_ = kv;
+    }
+
+    init();
+
+    CHECK_GT(intervalSeconds, 0)
+        << "intervalSeconds " << intervalSeconds << " should > 0!";
+    CHECK_GE(expiredSeconds, intervalSeconds)
+        << "expiredSeconds " << expiredSeconds
+        << " should >= intervalSeconds " << intervalSeconds;
+    CHECK(checkThread_.start());
+    checkThread_.addTimerTask(intervalSeconds * 1000,
+                              intervalSeconds * 1000,
+                              &ActiveHostsMan::cleanExpiredHosts,
+                              this);
+}
+
+void ActiveHostsMan::updateHostInfo(const HostAddr& hostAddr, const HostInfo& info) {
+    folly::RWSpinLock::ReadHolder rh(&lock_);
+    auto it = hostsMap_.find(hostAddr);
+    if (it == hostsMap_.end()) {
+        folly::RWSpinLock::UpgradedHolder uh(&lock_);
+        hostsMap_.emplace(std::move(hostAddr), std::move(info));
```

Comment:
If it is a new heartbeat, we should store it in kvstore as online.

Reply:
I add it in HBProcessor